runtime/bcutil/cfreader.c

```diff
@@ -448,6 +448,17 @@ readAttributes(J9CfrClassFile * classfile, J9CfrAttribute *** pAttributes, U_32
 				U_32 cursor = 0;
 				Trc_BCU_MalformedAnnotation(address);
 
+				/* Capture the errors with type_name_index & const_name_index in enum_const_value against the VM Spec */
+				if (BCT_ERR_INVALID_ANNOTATION_BAD_CP_INDEX_OUT_OF_RANGE == result) {
+					errorCode = J9NLS_CFR_ERR_BAD_INDEX__ID;
+					offset = address;
+					goto _errorFound;
+				} else if (BCT_ERR_INVALID_ANNOTATION_BAD_CP_UTF8_STRING == result) {
+					errorCode = J9NLS_CFR_ERR_BAD_NAME_INDEX__ID;
+					offset = address;
+					goto _errorFound;
```

Comment:
Does this behaviour apply across all supported JDK levels - 8, 11 & 15?  Does the RI report the errors for these cases in all releases?

Reply:
The rules for this in the Spec has been there since Java 7 as follows:
```
4.7.16 The RuntimeVisibleAnnotations attribute
4.7.16.1 The element_value structure
enum_const_value
The enum_const_value item is used if the tag item is e.
The enum_const_value item consists of the following two items:
type_name_index
The value of the type_name_index item must be a valid index into the constant_pool table. 
The constant_pool entry at that index must be a CONSTANT_Utf8_info structure (§4.4.7) 
representing a valid field descriptor (§4.3.2) that denotes the internal form of the binary name 
(§4.2.1) of the type of the enum constant represented by this element_value structure.
const_name_index
The value of the const_name_index item must be a valid index into the constant_pool table. 
The constant_pool entry at that index must be a CONSTANT_Utf8_info structure (§4.4.7) 
representing the simple name of the enum constant represented by this element_value structure.
```
Given the failing test case plus part of test framework code needs to be modified to the appropriate class version supported by the JDK, I can't simply verify it on Hotspot 8 and 11 but it works with Hotspot 15.
```
$ jdk15_hotspot/bin/java -version
openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)

test output:
java.lang.annotation.AnnotationFormatError: java.lang.IllegalArgumentException: Constant pool index out of bounds
```